fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/MetaDataProtoEditor.java

```diff
@@ -330,4 +367,13 @@ public static boolean hasUnion(@Nonnull Descriptors.FileDescriptor fileDescripto
         }
         return false;
     }
+
+    private static boolean hasField(@Nonnull DescriptorProtos.DescriptorProto.Builder builder, @Nonnull Descriptors.Descriptor messageType) {
+        for (DescriptorProtos.FieldDescriptorProto field : builder.getFieldList()) {
+            if (field.getTypeName().equals(messageType.getFullName())) {
```

Comment:
I think determining whether a field is of a given type is actually a little more complicated than this. Here's what I have in my work for #508:

```java
    private static boolean fieldHasType(@Nonnull DescriptorProtos.FileDescriptorProtoOrBuilder file,
                                        @Nonnull DescriptorProtos.FieldDescriptorProtoOrBuilder field,
                                        @Nonnull String typeName) {
        if (field.hasTypeName()) {
            if (field.getTypeName().startsWith(".")) {
                return field.getTypeName().equals(fullyQualifiedTypeName(file, typeName));
            } else {
                final String fieldTypeName = field.getTypeName();
                if (fieldTypeName.contains(".")) {
                    String[] parts = fieldTypeName.split("\\.");
                    if (!parts[parts.length - 1].equals(typeName)) {
                        return false;
                    }
                    if (!file.hasPackage() || file.getPackage().isEmpty()) {
                        return false;
                    }
                    String[] packageParts = file.getPackage().split("\\.");
                    int part = parts.length - 2;
                    int packagePart = packageParts.length - 1;
                    while (part >= 0 && packagePart >= 0) {
                        if (!packageParts[packagePart].equals(parts[part])) {
                            return false;
                        }
                        part--;
                        packagePart--;
                    }
                    return part < 0;
                } else {
                    return fieldTypeName.equals(typeName);
                }
            }
        } else {
            return false;
        }
    }
```

I'm not actually sure this is rock solid (as I haven't run this through as many tests as I would like), but this at least attempts to implement the name resolution rules in the [language guide](https://developers.google.com/protocol-buffers/docs/proto#packages).

Reply:
This is definitely better. where are you putting this? Probably we should share the code. I'll put a todo, and call your function once your code is in.

Reply:
Yeah, at the moment, my code is on my laptop, but will be a PR some day (I claim). I think a TODO works for now, and then I can update it when I submit my PR.